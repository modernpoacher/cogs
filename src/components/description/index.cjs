require('@babel/register')({ configFile: require.resolve('../../../babel.config.cjs') })

const debug = require('debug')

const log = debug('shinkansen-cogs/components/description')

log('`cogs` is awake')

module.exports = require('./index.jsx')
