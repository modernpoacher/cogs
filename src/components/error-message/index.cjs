require('@babel/register')({ configFile: require.resolve('../../../babel.config.cjs') })

const debug = require('debug')

const log = debug('shinkansen-cogs/components/error-message')

log('`cogs` is awake')

module.exports = require('./index.jsx')
