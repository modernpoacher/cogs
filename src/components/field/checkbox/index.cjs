require('@babel/register')({ ignore: [] })

const debug = require('debug')

const log = debug('shinkansen-cogs/components/field/checkbox')

log('`shinkansen` is awake')

const {
  default: component
} = require('./index.jsx')

module.exports = component
