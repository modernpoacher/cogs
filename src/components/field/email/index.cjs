require('@babel/register')({
  ignore: [
    /node_modules\/(?!shinkansen-)/
  ]
})

const debug = require('debug')

const log = debug('shinkansen-cogs/components/field/email')

log('`shinkansen` is awake')

const {
  default: component
} = require('./index.jsx')

module.exports = component
