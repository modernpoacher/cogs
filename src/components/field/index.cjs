require('@babel/register')({
  ignore: [
    /node_modules\/(?!shinkansen-)/
  ]
})

const debug = require('debug')

const log = debug('shinkansen-cogs/components/field')

log('`shinkansen` is awake')

/**
 *  Exports default, etc
 */
module.exports = require('./index.jsx')
