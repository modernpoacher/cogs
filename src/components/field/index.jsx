/**
 * Field component
 */
import React, { Component } from 'react'
import PropTypes from 'prop-types'

function onChange () {
  /* */
}

function onClick () {
  /* */
}

export default class Field extends Component {
  getDOMNode = () => this.domNode
  setDOMNode = (domNode) => !!(this.domNode = domNode) || delete this.domNode

  getClassName () {
    return 'input'
  }

  shouldComponentUpdate (props) {
    return (
      (props.id !== this.props.id) ||
      (props.required !== this.props.required) ||
      (props.disabled !== this.props.disabled) ||
      (props.readOnly !== this.props.readOnly) ||
      (props.tabIndex !== this.props.tabIndex) ||
      (props.accessKey !== this.props.accessKey) ||
      (props.placeholder !== this.props.placeholder) ||
      (props.onChange !== this.props.onChange)
    )
  }

  render () {
    const {
      id,
      required,
      disabled,
      readOnly,
      tabIndex,
      accessKey,
      placeholder
    } = this.props

    return (
      <input
        id={id}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        tabIndex={tabIndex}
        accessKey={accessKey}
        placeholder={placeholder}
        className={this.getClassName()}
        ref={this.setDOMNode}
      />
    )
  }
}

Field.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  required: PropTypes.bool,
  disabled: PropTypes.bool,
  readOnly: PropTypes.bool,
  tabIndex: PropTypes.number,
  accessKey: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func
}

Field.defaultProps = {
  required: false,
  disabled: false,
  readOnly: false,
  onChange
}

/**
 * ValueField component
 */
export class ValueField extends Field {
  getValue () {
    return this.getDOMNode()
      .value
  }

  setValue (value) {
    this.getDOMNode()
      .value = value
  }

  shouldComponentUpdate (props) {
    return (
      super.shouldComponentUpdate(props) ||
      (props.value !== this.props.value)
    )
  }

  handleChange = ({ target: { value } }) => {
    const { onChange } = this.props

    onChange(value)
  }
}

ValueField.propTypes = {
  ...Field.propTypes,
  value: PropTypes.string,
  defaultValue: PropTypes.string
}

ValueField.defaultProps = {
  ...Field.defaultProps
}

/**
 * CheckField component
 */
export class CheckField extends Field {
  getValue () {
    return this.getDOMNode()
      .checked
  }

  setValue (checked) {
    this.getDOMNode()
      .checked = checked
  }

  shouldComponentUpdate (props) {
    return (
      super.shouldComponentUpdate(props) ||
      (props.checked !== this.props.checked) ||
      (props.onClick !== this.props.onClick)
    )
  }

  handleClick = ({ target: { checked } }) => {
    const { onClick } = this.props

    onClick(checked)
  }

  handleChange = ({ target: { checked } }) => {
    const { onChange } = this.props

    onChange(checked)
  }
}

CheckField.propTypes = {
  ...Field.propTypes,
  checked: PropTypes.bool,
  defaultChecked: PropTypes.bool,
  onClick: PropTypes.func
}

CheckField.defaultProps = {
  ...Field.defaultProps,
  onClick
}
