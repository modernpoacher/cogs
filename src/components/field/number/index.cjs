require('@babel/register')({
  ignore: [
    /node_modules\/(?!shinkansen-)/
  ]
})

const debug = require('debug')

const log = debug('shinkansen-cogs/components/field/number')

log('`shinkansen` is awake')

const {
  default: component
} = require('./index.jsx')

module.exports = component
