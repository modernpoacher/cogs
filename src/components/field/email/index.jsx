/**
 * EmailField component
 */
import React from 'react'
import classnames from 'classnames'
import { ValueField } from 'shinkansen-cogs/components/field'

export default class EmailField extends ValueField {
  getClassName () {
    return classnames(super.getClassName(), 'email')
  }

  render () {
    const {
      id,
      name,
      value,
      defaultValue,
      required,
      disabled,
      readOnly,
      tabIndex,
      accessKey,
      placeholder
    } = this.props

    const className = this.getClassName()

    return (
      <input
        id={id}
        name={name}
        value={value}
        defaultValue={defaultValue}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        tabIndex={tabIndex}
        accessKey={accessKey}
        placeholder={placeholder}
        onChange={this.handleChange}
        className={className}
        type='email'
        ref={this.setField}
      />
    )
  }
}

EmailField.propTypes = {
  ...ValueField.propTypes
}

EmailField.defaultProps = {
  ...ValueField.defaultProps
}
