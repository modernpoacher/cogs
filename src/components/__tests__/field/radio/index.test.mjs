import React, { Component as mockComponent } from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import classnames from 'classnames'

import { CheckField } from 'shinkansen-cogs/components/field'
import Field from 'shinkansen-cogs/components/field/radio'

Enzyme.configure({ adapter: new Adapter() })

jest.mock('classnames', () => jest.fn(() => 'MOCK CLASSNAME'))

jest.mock('shinkansen-cogs/components/field', () => {
  class MockField extends mockComponent {
    getClassName () { }
  }

  return {
    __esModule: true,
    CheckField: class CheckField extends MockField { },
    default: MockField
  }
})

describe('shinkansen-cogs/components/field/radio', () => {
  describe('<Field />', () => {
    describe('With required props', () => {
      const component = (
        <Field
          id='MOCK ID'
          name='MOCK NAME'
          value='MOCK VALUE'
        />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('Prototype', () => {
        describe('`getClassName`', () => {
          it('is defined', () => {
            return expect(Field.prototype.getClassName)
              .toBeDefined()
          })
        })
      })

      describe('Instance', () => {
        let instance

        beforeEach(() => {
          instance = (
            shallow(component)
              .instance()
          )
        })

        describe('`handleClick`', () => {
          it('is defined', () => {
            return expect(instance.handleClick)
              .toBeDefined()
          })
        })

        describe('`handleChange`', () => {
          it('is defined', () => {
            return expect(instance.handleChange)
              .toBeDefined()
          })
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        const component = (
          <Field
            id='MOCK ID'
            name='MOCK NAME'
            title='MOCK TITLE'
            value='MOCK VALUE'
            required
            disabled
            readOnly
          />
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`getClassName()`', () => {
      let returnValue

      beforeEach(() => {
        jest.spyOn(CheckField.prototype, 'getClassName').mockReturnValue('MOCK GETCLASSNAME')

        const component = (
          <Field
            id='MOCK ID'
            name='MOCK NAME'
            value='MOCK VALUE'
          />
        )

        const instance = (
          shallow(component)
            .instance()
        )

        returnValue = instance.getClassName()
      })

      it('invokes `classnames`', () => {
        return expect(classnames)
          .toBeCalledWith('MOCK GETCLASSNAME', 'radio')
      })

      it('returns the classname', () => {
        return expect(returnValue)
          .toBe('MOCK CLASSNAME')
      })
    })

    describe('`handleClick()`', () => {
      it('invokes the `onClick` prop', () => {
        const MOCK_ONCLICK = jest.fn()

        const component = (
          <Field
            id='MOCK ID'
            name='MOCK NAME'
            value='MOCK VALUE'
            onClick={MOCK_ONCLICK}
          />
        )

        const instance = (
          shallow(component)
            .instance()
        )

        instance.handleClick({ target: { value: 'MOCK CHECKED' } })

        return expect(MOCK_ONCLICK)
          .toBeCalledWith('MOCK CHECKED')
      })
    })

    describe('`handleChange()`', () => {
      it('invokes the `onChange` prop', () => {
        const MOCK_ONCHANGE = jest.fn()

        const component = (
          <Field
            id='MOCK ID'
            name='MOCK NAME'
            value='MOCK VALUE'
            onChange={MOCK_ONCHANGE}
          />
        )

        const instance = (
          shallow(component)
            .instance()
        )

        instance.handleChange({ target: { value: 'MOCK CHECKED' } })

        return expect(MOCK_ONCHANGE)
          .toBeCalledWith('MOCK CHECKED')
      })
    })
  })
})
