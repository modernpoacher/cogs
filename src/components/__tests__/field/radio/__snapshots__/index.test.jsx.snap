// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shinkansen-cogs/components/field/radio <Field /> With additional props renders 1`] = `
<input
  className="MOCK CLASSNAME"
  disabled={true}
  id="MOCK ID"
  name="MOCK NAME"
  onChange={[Function]}
  onClick={[Function]}
  readOnly={true}
  required={true}
  type="radio"
  value="MOCK VALUE"
/>
`;

exports[`shinkansen-cogs/components/field/radio <Field /> With required props renders 1`] = `
<input
  className="MOCK CLASSNAME"
  id="MOCK ID"
  name="MOCK NAME"
  onChange={[Function]}
  onClick={[Function]}
  type="radio"
  value="MOCK VALUE"
/>
`;
